src/caching_key_wrapper.rs

```diff
@@ -11,17 +12,19 @@ use crate::key_provider::DataKey;
 use crate::KeyProvider;
 
 #[derive(ZeroizeOnDrop)]
-struct CachedEncryptionEntry {
+struct CachedEncryptionEntry<S: ArrayLength<u8>> {
     #[zeroize(skip)]
     created_at: Instant,
-    key: DataKey,
+    #[zeroize(skip)]
+    key: DataKey<S>,
     bytes_encrypted: usize,
     messages_encrypted: usize,
 }
 
 #[derive(ZeroizeOnDrop)]
-struct CachedDecryptionEntry {
-    key: Key<U16>,
+struct CachedDecryptionEntry<S: ArrayLength<u8>> {
+    #[zeroize(skip)]
+    key: Key<S>,
```

Comment:
Same here, when I add generic parameter `Key<S>` here, it stops working with ZeroizeOnDrop